modules/datagrid/72/launch/added/launch/infinispan-config.sh

```diff
@@ -208,6 +208,12 @@ function configure_infinispan_core() {
     fi
   fi
 
+  # If a cache with the name "default" exists, it must be the DEFAULT_CACHE, hence make sure it's first in the list
+  local filtered_cache_names=$(echo $CACHE_NAMES | sed -e 's/\bdefault\b/,/g' -e 's/,,//g')
+  if [ "$CACHE_NAMES" != "$filtered_cache_names" ]; then
+    CACHE_NAMES="default,$filtered_cache_names"
+  fi
```

Comment:
If somebody defines a cache to use the name `default` and it's not in cache names, do we need to add it to CACHE_NAMES?  For example where MEMCACHED_CACHE was being set to default?  Is it already a requirement that all caches named for specific cache types be listed in CACHE_NAMES?

Reply:
@rcernich You're right, however [this line](https://github.com/ryanemerson/jboss-datagrid-7-openshift-image/blob/bf397f3a914abf0eadb2fc9dafd265f91bcd7e45/modules/datagrid/72/launch/added/launch/infinispan-config.sh#L198) already ensures that CACHE_NAMES is populated with "default" if not defined.

Reply:
...and what if the user specifies:

* `CACHE_NAMES=memcached`
* `MEMCACHED_CACHE=memcached`

Sorry to be dense, I don't understand the relationship between the _CACHE vars and CACHE_NAMES.  Is it that MEMCACHED_CACHE specifies the name of the cache that should be exposed through the memcached interface, i.e. CACHE_NAMES configures the list of caches and the other _CACHE settings define which of those are exposed over which protocols?  In which case, your change simply ensures that if `default` is specified that it ends up being at the front of the list, right?

Reply:
No worries, all valid questions.

>Is it that MEMCACHED_CACHE specifies the name of the cache that should be exposed through the memcached interface

Yes

> CACHE_NAMES=memcached
> MEMCACHED_CACHE=memcached

In that scenario only one cache, "memcached" is created and the "memcached-connector" points to that cache. Furthermore, the "memached" cache is the default for the container. 


> In which case, your change simply ensures that if default is specified that it ends up being at the front of the list, right?

Yes exactly.